### ISSUE TITLE
Autoplay not working for youtube videos

### ISSUE THREAD

**Issue by Johannes5** [sampotts/plyr#2537]
### Expected behaviour
Video auto-plays when the component mounts.

The specific flow I have in mind:
user clicks on thumbnail -> a route like /video/youtube/38j9hF378hf opens -> video is playing (in fullscreen, if possible) automatically

### Actual behaviour
No auto-play for youtube videos.
Autoplay for vimeo (only when enabled by browser)

### Environment

- Browser: tried it on safari, firefox, or chromium browsers
- Version: I'm using "plyr-react": "^5.1.0",
- Operating System: MacOS (using M1 chip)
- Version: 12.3.1

----
I'm using plyr-react and I was advised to forward the issue here:
https://github.com/chintan9/plyr-react/issues/968

Basically, **I'm calling the play method on the "canplay" event.**
And autoplay doesn't trigger for youtube videos (when the route is loaded) but the video can sometimes start playing at a (seemingly) random in the background.
